src/components/Subthoughts.tsx

```diff
@@ -77,6 +77,10 @@ if (!toggleContextViewShortcut) throw new Error('toggleContextView shortcut not
 
 const PAGINATION_SIZE = 100
 
+/** Check if the given path is a leaf. */
+const isLeaf = (state: State, path: Path) => !getAllChildren(state, pathToContext(path))
+  .some((child: Child) => !isFunction(child.value))
```

Comment:
I'm not sure how this logic was working before, but it doesn't take into account `showHiddenThoughts`. You can use the `hasChildren` predicate selector instead:

https://github.com/cybersemics/em/blob/5101be65c855fe7e1a18adeb4efd62183d12c6c5/src/selectors/getChildren.ts#L46-L52